eth-contracts/contracts/service/ClaimFactory.sol

```diff
@@ -1,70 +1,173 @@
-    return (fundingAmount);
-  }
-
-  function initiateClaim() external {
-    require(
-      block.number - lastClaimBlock > claimBlockDiff,
-      'Required block difference not met');
-
-    bool minted = audiusToken.mint(address(this), fundingAmount);
-    require(minted, 'New tokens must be minted');
-
-    // Approve token transfer to staking contract address
-    audiusToken.approve(stakingAddress, fundingAmount);
-
-    // Fund staking contract with proceeds
-    Staking stakingContract = Staking(stakingAddress);
-    stakingContract.fundNewClaim(fundingAmount);
-
-    // Increment by claim difference
-    // Ensures funding of claims is repeatable given the right block difference
-    lastClaimBlock = lastClaimBlock + claimBlockDiff;
-  }
+contract ClaimFactory is RegistryContract {
+    using SafeMath for uint256;
+    RegistryInterface registry = RegistryInterface(0);
+    // standard - imitates relationship between Ether and Wei
+    uint8 private constant DECIMALS = 18;
+
+    address tokenAddress;
+    bytes32 stakingProxyOwnerKey;
+    bytes32 serviceProviderFactoryKey;
+
+    // Claim related configurations
+    uint fundRoundBlockDiff = 10;
+    uint fundBlock = 0;
+
+    // 20 AUD
+    // TODO: Make this modifiable based on total staking pool?
+    uint fundingAmount = 20 * 10**uint256(DECIMALS); // 100 * 10**uint256(DECIMALS);
+
+    // Denotes current round
+    uint roundNumber = 0;
+
+    // Total claimed so far in round
+    uint totalClaimedInRound = 0;
+
+    // Staking contract ref
+    ERC20Mintable internal audiusToken;
+
+    event RoundInitiated(
+      uint _blockNumber,
+      uint _roundNumber,
+      uint _fundAmount
+    );
+
+    event ClaimProcessed(
+      address _claimer,
+      uint _rewards,
+      uint _oldTotal,
+      uint _newTotal
+    );
+
+    constructor(
+      address _tokenAddress,
+      address _registryAddress,
+      bytes32 _stakingProxyOwnerKey,
+      bytes32 _serviceProviderFactoryKey
+    ) public {
+        tokenAddress = _tokenAddress;
+        stakingProxyOwnerKey = _stakingProxyOwnerKey;
+        serviceProviderFactoryKey = _serviceProviderFactoryKey;
+        audiusToken = ERC20Mintable(tokenAddress);
+        registry = RegistryInterface(_registryAddress);
+        fundBlock = 0;
+    }
+
+    function getFundingRoundBlockDiff()
+    external view returns (uint blockDiff)
+    {
+        return fundRoundBlockDiff;
+    }
+
+    function getLastFundBlock()
+    external view returns (uint lastFundBlock)
+    {
+        return fundBlock;
+    }
+
+    function getFundsPerRound()
+    external view returns (uint amount)
+    {
+        return fundingAmount;
+    }
+
+    function getTotalClaimedInRound()
+    external view returns (uint claimedAmount)
+    {
+        return totalClaimedInRound;
+    }
+
+    // Start a new funding round
+    // TODO: Permission caller to contract deployer or governance contract
+    function initiateRound() external {
```

Comment:
v1 - Confirm and permission if needed. Consider restricting to governance, deployer, and known SPs